File: ch1/fun.f90
program main
implicit none
integer n
parameter (n=11)
real(kind=8), dimension(n)::x
real(kind=8), dimension(n)::y
integer i

!linspace(-1,1,11)
do i=1,11
  x(i) = -1.0 + 2.0*(i-1)/(n-1)
end do

!evaluate using vector operations
y = max(x*x-2*x,0.)
print "(a7,f6.3,a8,f6.3)","f(-1) =",y(1), ", f(1) =",y(n)
end

File: ch1/fun.pas
program Fun;
const
  N = 11;

(* custom function *)
function fun(x: real): real;
var
   (* local variable declaration *)
   z: real;
begin
   z := x*x - 2*x;
   if (z >= 0) then
      fun := z
   else
      fun := 0;
end;

(* variable declarations *) 
type
   vec11 = array [0..N] of real;
var
   y: vec11;
   i : integer;
   x : real;

(* main program *) 
begin
  for i :=0 to N do
    begin
      x := -1.0 + 2*i/(N-1);
      y[i] := fun(x);
    end;
  writeln ('y(-1)=',y[0]:5:2,', y(1)=',y[N-1]:5:2);
end.



